From 26e8fe568123cc957ea500c4b747f44fa8193edc Mon Sep 17 00:00:00 2001
From: Gleb Baryshev <gleb.baryshev@gmail.com>
Date: Sun, 13 Aug 2017 21:37:52 +0300
Subject: [PATCH] Fix KF5 issues

#33 Library has wrong versioning (libbasketcommon.so.SOVERSION)
#34 Translations are not properly used
#35 Documentation still requires kdelibs4support for dtd/ktex.dtd
---
 CMakeLists.txt       | 3 +++
 doc/en/index.docbook | 4 ++--
 src/CMakeLists.txt   | 4 ++--
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2997365e..7b54ace3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,9 @@ ENDIF(GPGME_FOUND)
 
 #TODO: find meinproc
 
+#Make libbasketcommon search for translations in basket.mo
+ADD_DEFINITIONS(-DTRANSLATION_DOMAIN=\"basket\")
+
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h )
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 
diff --git a/doc/en/index.docbook b/doc/en/index.docbook
index ddef489a..586014e8 100644
--- a/doc/en/index.docbook
+++ b/doc/en/index.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
   <!ENTITY basket "<application>BasKet</application>">
   <!ENTITY % addindex "IGNORE">
   <!ENTITY % English "INCLUDE" > <!-- change language only here -->
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9ea886d5..f4cf523f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -59,7 +59,7 @@ IF(HAVE_BALOO)
   list(INSERT basketcommon_LIB_SRCS 10 nepomukintegration.cpp)
 ENDIF(HAVE_BALOO)
 
-qt5_wrap_ui(basket_FORM_HDRS passwordlayout.ui basketproperties.ui settings_versionsync.ui)
+ki18n_wrap_ui(basket_FORM_HDRS passwordlayout.ui basketproperties.ui settings_versionsync.ui)
 
 QT5_ADD_DBUS_ADAPTOR(basketcommon_LIB_SRCS org.basket.BNPView.xml bnpview.h BNPView)
 
@@ -93,7 +93,7 @@ IF(HAVE_BALOO)
   target_link_libraries(basketcommon KF5::BalooFiles)
 ENDIF(HAVE_BALOO)
 
-set_target_properties(basketcommon PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
+set_target_properties(basketcommon PROPERTIES VERSION ${Qt5Core_VERSION} SOVERSION ${Qt5Core_VERSION_MAJOR})
 
 install(TARGETS basketcommon DESTINATION ${LIB_INSTALL_DIR})
 
